charmhelpers/core/hookenv.py

```diff
@@ -1088,22 +1097,31 @@ def function_tag():
     return os.environ.get('JUJU_FUNCTION_TAG') or action_tag()
 
 
-def status_set(workload_state, message):
+def status_set(workload_state, message, application_status=False):
     """Set the workload state with a message
 
     Use status-set to set the workload state with a message which is visible
     to the user via juju status. If the status-set command is not found then
     assume this is juju < 1.23 and juju-log the message instead.
 
-    workload_state -- valid juju workload state.
-    message        -- status update message
+    workload_state     -- valid juju workload state. str or WL_STATES
+    message            -- status update message
+    application_status -- Whether this is an application state set
     """
-    valid_states = ['maintenance', 'blocked', 'waiting', 'active']
-    if workload_state not in valid_states:
+    # Extract the value if workload_state is an Enum
+    try:
+        workload_state = workload_state.value
+    except AttributeError:
+        pass
+    workload_state = workload_state.lower()
+    if workload_state not in [s.lower() for s in WL_STATES.__members__.keys()]:
```

Comment:
I recommend implementing `__contains__()` within the enum itself. All of the complexity could then be re-used without becoming distracting.

Reply:
I don't think that is the right thing to do. __contains__ is already implemented in enum.EnumMeta and correctly checks if an Enum is a member of the Enum class. What you are proposing seems to be making __contains__ handle strings rather than Enums which feels wrong.